server/controllers/posts.js

```diff
@@ -25,7 +26,35 @@ const savePost = asyncMiddleware(async (req, res, next) => {
   res.status(200).json({ success: true })
 })
 
+
+const searchPosts = function(req, res) {
+  if (!req.query.tag) {
+    res.status(400).send('no tag specified!')
+    return
+  }
+  const value  = req.query.tag
+
+  let postsDocsPromise = searchService.findByTag(value)
+  postsDocsPromise.then(snapshot => {
+    if (snapshot.empty) {
+      res.send(400).send('No matching documents.')
```

Comment:
if you write here return res.send you don't need to use else
```suggestion
      if(snapshot.empty) {
        return res.send(400).send('No matching documents.')
     }
     // rest of your code without else statement
```

Reply:
done